src/js/utils/JSONUtil.js

```diff
@@ -155,23 +154,31 @@ module.exports = {
         case 'begin-object':
           // Keep track of keyed objects
           if (objectKey) {
-            keyLines.push({
+            let blockToken = {
```

Comment:
I am open to suggestions, since I am currently focused on another task. From a quick look that I had it's a bit tricky to extract a re-usable function without carrying almost every variable in scope along...